### DISCUSSION TITLE
Mouse Mover (Python Script + ScriptKit)

### DISCUSSION THREAD

**Discussion by benjaminmodayil** [johnlindquist/kit#467] [Share]
😅 This one is more of a Python script, but I use ScriptKit to launch it. There are moments where I have to leave my computer to address something in another room or need more time than the allotted 10 minutes that Slack provides before it makes it look like I'm unavailable. For the most part, even if I'm in another room, I am still available for conversation/inquiries, so I don't want my coworkers to think I can't be bothered if my status looks like I'm away, but in reality I just might be eating/cooking. So this keeps my computer + work messaging app status in the green so it doesn't look like I'm unavailable.

The Python script requires `pyautogui` and `time` modules. I'm not the best for Python troubleshooting as I just learned some Python through YouTube tutorials. All I remember from when I wrote this was that it was a pain to get my computer using the latest version of Python.

From the Python script below, `pyautogui` lets you __exit the script from taking control of your mouse by running `ctrl + c` or moving your mouse to the top right corner of your screen__. Because you can't run `ctrl + c` into a non-existent terminal when using it through ScriptKit, you need to move your mouse to the top right corner of your monitor to exit. 

**Note:** Before connecting/disconnecting external displays while the script is active, you must exit the script first, otherwise you'll have to reconnect/disconnect your displays to the condition of your computer when you started the script to exit.

You'll have to squint to see the mouse once the script starts + I only run it for three seconds before closing... Sorry about that.

https://user-images.githubusercontent.com/12306850/135878354-7ab7e798-c63d-4a5e-8ee8-78e938d66b9a.mp4

```python

#! /usr/bin/env python3
import pyautogui
import time

while True:
    pyautogui.moveRel(0, 50, duration=1.5)
    time.sleep(1)
    pyautogui.moveRel(0, -50, duration=1.5)
    time.sleep(1)
    pyautogui.moveRel(50, 0, duration=1.5)
    time.sleep(1)
    pyautogui.moveRel(-50, 0, duration=1.5)
    time.sleep(1)

# You can remove a bunch of the repetition above. I just like seeing the mouse move a lot to know it's working.

```

Here's the ScriptKit launcher. This will vary depending on your Python path + Python script path.

```js
// Menu: Mouse Mover
// Description: Moves your mouse so you don't go inactive in Slack or Microsoft Teams
// Author: Benjamin Modayil
// Twitter: @24props

await exec(`/PATH-TO-PYTHON-EXECUTABLE/python3.9 /PATH-TO-SCRIPT/i-am-here.py`)
```

**Comment by johnlindquist** [comment]
Awesome! Thanks so much for sharing a video too. I don't think too many people realize you can launch whatever you want (python, etc) using Script Kit. Glad you figured it out!

Couple thoughts:

1. There's a JS lib called "robotjs" that I _think_ could accomplish the same thing. @gm90 might know more:

https://github.com/johnlindquist/kit/discussions/288

2. I'm considering adding a `python` helper that would look like this:

```js
// I may make something like this a global fn
let python = async ([scriptContent]) => {
  let scriptPath = tmpPath("hello-world.py")
  await outputFile(scriptPath, scriptContent)

  await $`python ${scriptPath}`
}

// So all you would need to do is this:
await python`

print("Hello world")

`
```

**Reply by benjaminmodayil** [reply]
Oooh. I'll have to dig into `robotjs`. Thanks for the suggestion!

A Python helper would be cool too. I like the idea of bringing in the script to one Kit file versus keeping separate files.

**Reply by gm90** [reply]
Yh robotjs did the business for me, (good memory @johnlindquist 👏 ). Reminds me of an app colleagues used to use at work, wiggle on windows. Jiggler for mac. Sounds more fun to build it yourself though ;)
